test/MessageTest.php

```diff
@@ -221,4 +224,177 @@ public function testFromStringWithCrlfAndRfc2822FoldedHeaders()
         $this->assertEquals(1, count($parts));
         $this->assertEquals('attachment; filename="test.txt"', $parts[0]->getDisposition());
     }
+
+    public function testDecodeMultipartMimeMessageWithMessagePartAlternatives()
+    {
+        $rawMessage = <<<EOL
+--001a11447dc881e40f0537fe6d5a
+Content-Type: multipart/alternative; boundary=001a11447dc881e40b0537fe6d58
+
+--001a11447dc881e40b0537fe6d58
+Content-Type: text/plain; charset=UTF-8
+
+This is a test email with 1 attachment.
+
+--001a11447dc881e40b0537fe6d58
+Content-Type: text/html; charset=UTF-8
+Content-Transfer-Encoding: quoted-printable
+
+<div dir=3D"ltr">This is a test email with 1 attachment.<br clear=3D"all"><=
+div><br></div>-- <br><div class=3D"gmail_signature" data-smartmail=3D"gmail=
+_signature"><div dir=3D"ltr"><img src=3D"https://sendgrid.com/brand/sg-logo=
```

Comment:
Use an example.com domain instead to keep it agnostic and to prevent somebody from copying and sending this.

Reply:
Works for me. Will do.